metaflow/plugins/tagging_cli.py

```diff
@@ -0,0 +1,424 @@
+    else:
+        # In quiet mode, we display things much more simply so it is parseable.
+        # When displaying by run, we output three columns:
+        #  - the comma separated list of runs (usually 1 but more if --all-runs for eg)
+        #  - the comma separated list of system tags
+        #  - the comma separated list of user tags
+        # Columns are separated by a semicolon surrounded by a space on each side
+        # to make it visually clear when there are no system tags for example
+        obj.echo_always(' ; '.join([
+            cat,
+            ','.join(sorted(system_tags)),
+            ','.join(sorted(all_tags.difference(system_tags)))]))
+
+
+def tag_category_format(obj, cat, runs, is_system):
+    if not obj.is_quiet:
+        # Pretty printing computations
+        max_length = max([len(t) for t in runs] if runs else [0])
+
+        all_runs = sorted(runs)
+
+        num_runs = len(all_runs)
+
+        # We consider 4 spaces in between column and we consider 120 characters total
+        # width and we want 120 >= column_count*max_length + (column_count - 1)*4
+        column_count = 124 // (max_length + 4)
+        if column_count == 0:
+            # Make sure we have at least 1 column even for very very long tags
+            column_count = 1
+            words_per_column = num_runs
+        else:
+            words_per_column = (num_runs + column_count - 1) // column_count
+
+        # Extend all_runs by empty runs to be able to easily fill up the lines
+        addl_runs = words_per_column * column_count - num_runs
+        if addl_runs > 0:
+            all_runs.extend([' ' for _ in range(addl_runs)])
+
+        lines = []
+        lines.append('For tag %s' % (cat if is_system else '*%s*' % cat))
+        for i in range(words_per_column):
+            line_values = [all_runs[i+j*words_per_column] for j in range(column_count)]
+            length_values = [max_length for l in line_values]
+            formatter = "    ".join(["{:<%d}" % l for l in length_values])
+            lines.append(formatter.format(*line_values))
+        lines.append("")
+        obj.echo('\n'.join(lines), err=False)
+    else:
+        # In quiet mode, we display things much more simply so it is parseable.
+        # When displaying by tag, we output twp columns:
+        #  - the tag (user:<tag> or sys:<tag>)
+        #  - the comma separated list of runs
+        # Columns are separated by a semicolon surrounded by a space on each side
+        # to make it consistent with the by_run listing
+        obj.echo_always(' ; '.join([
+            'system:%s' % cat if is_system else 'user:%s' % cat,
+            ','.join(runs)]))
+
+
+def print_tags(obj, metaflow_run_client):
+    system_tags = {metaflow_run_client.pathspec: metaflow_run_client.system_tags}
+    all_tags = {metaflow_run_client.pathspec: metaflow_run_client.tags}
+    return format_tags(obj, system_tags, all_tags, by_tag=False)
+
+
+def is_prod_token(token):
```

Comment:
I am not sure if this is the right check. The `@project` decorator introduces the notion of a `production` branch as well and it is easy for future code committers to get confused about what we mean here. I think we should consider allowing the end users to modify SFN run tags.

Reply:
@jasonge27 : what was the rationale for preventing this?

Reply:
The original intention was to avoid modifying tags for deployed runs since the ownership of deployed runs is not clear. Agree that this is probably not the best way to check if the run is deployed on SFN/Meson

Reply:
This requires a bit more thought. I think one way we could modify it is by allowing tags to be specified in the triggers. This would make ownerhsip a bit clearer.

Reply:
Given that we don't have any notion of `prod_token`, we should remove this check entirely and rely on user being in the correct namespace.